Gemfile

```
@@ -7,12 +7,12 @@ rubocop_version = '= 1.25.0'
 
 chef_vault_version = '~> 4.0'
 
-chef_version = if Bundler.current_ruby.on_26?
-                 '= 15.8.23'
-               elsif Bundler.current_ruby.on_27?
+chef_version = if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.7.0') && Gem::Version.new(RUBY_VERSION) < Gem::Version.new('3.0.0')
                  '= 16.17.18'
-               else
+               elsif Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.0.0') && Gem::Version.new(RUBY_VERSION) < Gem::Version.new('3.1.0')
                  '= 17.10.0'
+               elsif Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.1.0')
+                 '= 18.1.0'
```

I think this was slightly easier to read using `Bundler.current_ruby`. Why the switch to Gem::Version?
```suggestion
chef_version = if Bundler.current_ruby.on_27?
                 '= 16.17.18'
               elsif Bundler.current_ruby.on_30?
                 '= 17.10.0'
               elsif Bundler.current_ruby.on_31?
                 '= 18.1.0'
               end
```

---

@gravesb, Looks like `on_31?` isn't supported with the bundler that is being used in the builds, [v2.3.3](https://app.travis-ci.com/github/cerner/cerner_splunk/jobs/594381424#L294) but was added to bundler [v2.3.12](https://github.com/rubygems/rubygems/blob/master/bundler/CHANGELOG.md#2312-april-20-2022)

---

Can you go ahead and update bundler then? Even the latest bundler only requires ruby >= 2.6. The breaking changes in 2.4 don't look like they'll affect this project.

---

Using an else instead of elsif on that last qualifier is fine too.